src/compiler/checker.ts

```diff
@@ -30445,8 +30445,31 @@ namespace ts {
             const functionIsUsedInBody = forEachChild(ifStatement.thenStatement, function check(childNode): boolean | undefined {
                 if (isIdentifier(childNode)) {
                     const childSymbol = getSymbolAtLocation(childNode);
-                    if (childSymbol && childSymbol.id === testedFunctionSymbol.id) {
-                        return true;
+                    if (childSymbol && childSymbol === testedFunctionSymbol) {
+                        // If the test was a simple identifier, the above check is sufficient
+                        if(isIdentifier(ifStatement.expression)) {

```

Comment:
Formatting in multiple places here is missing space between control flow keywords and `(`

Reply:
Shouldn't the linter have caught this ? The lint is clean though 😕